1. Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks): zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110], ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]. Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны. Найдите коэффициент корреляции Пирсона с помощью ковариации и среднеквадратичных отклонений двух признаков, а затем с использованием функций из библиотек numpy и pandas.

In [1]:
zp <- c(35, 45, 190, 200, 40, 70, 54, 150, 120, 110)
ks <- c(401, 574, 874, 919, 459, 739, 653, 902, 746, 832)

    Сначала найдём средние значения величин. Буду пользоваться программными средствами языка для нахождения промежуточных значений.

In [2]:
mean_zp <- mean(zp)
mean_ks <- mean(ks)

    Найдем верхнюю границу доверительного интервала:

In [3]:
cov_zp_ks <- mean(zp*ks) - mean_zp * mean_ks
cov_zp_ks

[1] 9157.84

    Найдём ковариацию программными средствами:

In [4]:
cov(zp, ks)

[1] 10175.38

    Значения не совпадают. Сразу было найдена проблема несовпадения. В R функция cov даёт смещенную оценку. В Python это легко устраняется атрибутом ddof, в то время как в R я ничего подобного не нашёл. 

    В Python результат 9157.84 выдаёт строка: np.cov(zp, ks, ddof=0)[0][1] 

    Приведу к смещенной оценке cov:

In [5]:
cov_zp_ks <- cov_zp_ks * 10 / 9
c(cov_zp_ks, cov(zp, ks))

[1] 10175.38 10175.38

    Теперь, как видно, значения стали равны. Поэтому можем перейти к следующему этапу.

    Найдём коэффициент корреляции Пирсона:

In [6]:
r <- cov_zp_ks / (sd(zp) * sd(ks))
r

[1] 0.8874901

    В языке R корреляция Пирсона представлена функцией cor, поэтому буду использовать её.

In [7]:
c(cor(zp, ks), r)

[1] 0.8874901 0.8874901

    Как видно, получившиеся значения совпадают.

2. Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:  131, 125, 115, 122, 131, 115, 107, 99, 125, 111. Известно, что в генеральной совокупности IQ распределен нормально. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

    Создаём массив данных измерений и записываем в соответствующие значения необходимые показатели выборки:

In [8]:
x <- c(131, 125, 115, 122, 131, 115, 107, 99, 125, 111)
m <- mean(x)
sigma <- sd(x)
n <- 10

    Найдём обратное значение функции стандартного нормального распределения по таблице Стьюдента. При количестве степеней свободы 9 и уровнем значимости 0.05 равно 2.262.

In [9]:
c <- 2.262

    Найдем верхнюю границу доверительного интервала:

In [10]:
t_1 <- m + sigma * c / sqrt(n)
t_1

[1] 125.6434

    Найдём нижнюю границу доверительного интервала:

In [11]:
t_2 <- m - sigma * c / sqrt(n)
t_2

[1] 110.5566

    Значит с надежностью в 0.95 мы можем сказать, что величина Х находится в данном интервале:

In [12]:
c(t_2, t_1)

[1] 110.5566 125.6434

3.  Известно, что рост футболистов в сборной распределен нормально с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27, среднее выборочное составляет 174.2. Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [13]:
sigma <- 25
gamma <- 0.95
m <- 174.2
n <- 27

    Сначала найдём обратное значение функции стандартного нормального распределения.

In [14]:
c <- qnorm((gamma + 1) / 2)
c

[1] 1.959964

    Найдем верхнюю границу доверительного интервала:

In [15]:
t_1 <- m + sigma * c / sqrt(n)
t_1

[1] 183.6299

    Найдём нижнюю границу доверительного интервала:

In [16]:
t_2 <- m - sigma * c / sqrt(n)
t_2

[1] 164.7701

    Доверительный интервал равен:

In [17]:
c(t_2, t_1)

[1] 164.7701 183.6299